hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/states/endpoint/VersionEndpointTask.java

```diff
@@ -135,4 +145,27 @@ private void checkVolumeSet(MutableVolumeSet volumeSet,
       volumeSet.writeUnlock();
     }
   }
+
+  private void cleanupDeletedContainer(HddsVolume volume) {
```

Comment:
Lets move the startup cleanup steps somewhere inside `ContainerReader` so that they can be done as part of the initial iteration that builds up the container set with one thread per volume, instead of doing a second iteration of all containers in one thread afterwards. Probably inside `ContainerRead#verifyAndFixupContainerData` or `KeyValueContainerUtil#parseKVContainerData`.